LibreNMS/RRD/RrdDefinition.php

```diff
@@ -50,13 +51,20 @@ public static function make()
      * @param int $max Maximum allowed value.  null means undefined.
      * @param int $heartbeat Heartbeat for this dataset. Uses the global setting if null.
      * @return $this
+     * @throws InvalidRrdNameException
      */
     public function addDataset($name, $type, $min = null, $max = null, $heartbeat = null)
     {
         global $config;
+        $name = $this->escapeName($name);
+
+        if (strlen($name) == 0 || strlen($name) > 19) {
```

Comment:
I agree that > 19 should probably never happen, but I put the check in, in the case that code below changes at some point. I can remove if you want.